src/vm/codeman.cpp

```diff
@@ -1304,67 +1305,109 @@ void EEJitManager::SetCpuInfo()
         // and edx in buffer[12-15].
+        
         // We will set the following flags:
+        // CORJIT_FLAG_USE_SSE2 is required
+        //    SSE       - EDX bit 25    (buffer[15] & 0x02)
+        //    SSE2      - EDX bit 26    (buffer[15] & 0x04)
         // CORJIT_FLAG_USE_SSE3 if the following feature bits are set (input EAX of 1)
-        //    SSE3 - ECX bit 0     (buffer[8]  & 0x01)
+        //    CORJIT_FLAG_USE_SSE2
+        //    SSE3      - ECX bit 0     (buffer[8]  & 0x01)
         // CORJIT_FLAG_USE_SSSE3 if the following feature bits are set (input EAX of 1)
-        //    SSE3 - ECX bit 0     (buffer[8]  & 0x01)
-        //    SSSE3 - ECX bit 9    (buffer[9]  & 0x02)
+        //    CORJIT_FLAG_USE_SSE3
+        //    SSSE3     - ECX bit 9     (buffer[9]  & 0x02)
         // CORJIT_FLAG_USE_SSE41 if the following feature bits are set (input EAX of 1)
-        //    SSE3 - ECX bit 0     (buffer[8]  & 0x01)
-        //    SSSE3 - ECX bit 9    (buffer[9]  & 0x02)
-        //    SSE4.1 - ECX bit 19  (buffer[10] & 0x08)
+        //    CORJIT_FLAG_USE_SSSE3
+        //    SSE4.1    - ECX bit 19    (buffer[10] & 0x08)
         // CORJIT_FLAG_USE_SSE42 if the following feature bits are set (input EAX of 1)
-        //    SSE3 - ECX bit 0     (buffer[8]  & 0x01)
-        //    SSSE3 - ECX bit 9    (buffer[9]  & 0x02)
-        //    SSE4.2 - ECX bit 20  (buffer[10] & 0x10)
+        //    CORJIT_FLAG_USE_SSE41
+        //    SSE4.2    - ECX bit 20    (buffer[10] & 0x10)
         // CORJIT_FLAG_USE_POPCNT if the following feature bits are set (input EAX of 1)
-        //    SSE3 - ECX bit 0     (buffer[8]  & 0x01)
-        //    SSSE3 - ECX bit 9    (buffer[9]  & 0x02)
-        //    SSE4.2 - ECX bit 20  (buffer[10] & 0x10)
-        //    POPCNT - ECX bit 23  (buffer[10] & 0x80)
+        //    CORJIT_FLAG_USE_SSE42
+        //    POPCNT    - ECX bit 23    (buffer[10] & 0x80)
         // CORJIT_FLAG_USE_AVX if the following feature bits are set (input EAX of 1), and xmmYmmStateSupport returns 1:
-        //    OSXSAVE - ECX bit 27 (buffer[11] & 0x08)
-        //    AVX - ECX bit 28     (buffer[11] & 0x10)
+        //    CORJIT_FLAG_USE_SSE42
+        //    OSXSAVE   - ECX bit 27   (buffer[11] & 0x08)
+        //    XGETBV    - XCR0[2:1]    11b
+        //    AVX       - ECX bit 28   (buffer[11] & 0x10)
         // CORJIT_FLAG_USE_FMA if the following feature bits are set (input EAX of 1), and xmmYmmStateSupport returns 1:
-        //    FMA - ECX bit 12     (buffer[9]  & 0x10)
+        //    CORJIT_FLAG_USE_AVX
+        //    FMA       - ECX bit 12   (buffer[9]  & 0x10)
         // CORJIT_FLAG_USE_AVX2 if the following feature bit is set (input EAX of 0x07 and input ECX of 0):
-        //    AVX2 - EBX bit 5     (buffer[4]  & 0x20)
+        //    CORJIT_FLAG_USE_AVX
+        //    AVX2      - EBX bit 5    (buffer[4]  & 0x20)
         // CORJIT_FLAG_USE_AVX_512 is not currently set, but defined so that it can be used in future without
```

Comment:
AVX-512, if/when support is added is interesting, as it requires both AVX2 and FMA support (which both depend on AVX, but not on eachother).